_episodes/l1-06-releases_tags.md

```diff
@@ -3,19 +3,355 @@ title: "Code versions, releases and tags"
+colleague), remember that you can always obtain the current commit hash like so:
+
+~~~sh
+$ git rev-parse --short HEAD
+~~~
+{: .commands}
+~~~
+047e4fe
+~~~
+{: .output}
+
+### When should I consider creating releases?
+
+You *might* consider creating releases if:
+
+- You have a large, distributed user base and you don't want them just using some
+  random commit from your repository
+- You periodically add or change features in a non-backwards-compatible way
+- You want to bundle these features together and advertise them to your users
+- Your software is complex and you can't guarantee that every commit on `main` has been
+  thoroughly tested
+- You want to be able to support several versions of your software simultaneously (e.g.
+  you intend to maintain a v2 of your software, even after v3 has been released)
+- You want to provide a place to download files that you've generated (such as `.exe`
+  files) and distribute them to users
+
+In general, though, creating a release is also just a convenient way to label a
+particular commit, so it's a useful way to ensure that the users you're sharing it with
+-- who may not be technically savvy -- are using a specific commit, rather than simply
+whatever the latest commit on `main` is.
+
+## Labelling a particular commit with `git tag`
+
+Git tags provide a way to give human-readable names to a specific commits. We will now
+go through how to add and remove tags to your repository.
+
+Firstly, remind yourself what the history for your `recipe` repository looks like with
+`git log`. Mine looks like this:
+
+~~~
+$ git log
+~~~
+{: .commands}
+~~~
+commit 09c9b3ba757d0073d021bbaeba809a55b6c78d8e
+Author: Diego Alonso Álvarez <d.alonso-alvarez@imperial.ac.uk>
+Date:   Tue Nov 15 14:01:47 2022 +0000
+
+    Revert "Added instruction to enjoy"
+
+    This reverts commit 1171d948be795eed6d4fe79d51e6889aed7f5821.
+
+commit 366f4b51e661fd96a9bde001138886a4216b7da1
+Author: Diego Alonso Álvarez <d.alonso-alvarez@imperial.ac.uk>
+Date:   Tue Nov 15 13:55:39 2022 +0000
+
+    Added 1/2 onion to ingredients
+
+commit 1171d948be795eed6d4fe79d51e6889aed7f5821
+Author: Diego Alonso Álvarez <d.alonso-alvarez@imperial.ac.uk>
+Date:   Tue Nov 15 13:55:16 2022 +0000
+
+    Added instruction to enjoy
+
+commit 6ff8aa58e55545956cf816baf676a17aeb74d993
+Author: Diego Alonso Álvarez <d.alonso-alvarez@imperial.ac.uk>
+Date:   Tue Nov 15 13:51:55 2022 +0000
+
+    adding ingredients and instructions
+~~~
+{: .output}
+
+(Note that yours may look different depending on whether you followed the steps yourself
+or downloaded the pre-made repository.)
+
+Let's say that you have decided that the point at which you added half an onion was a
+highpoint in the recipe's history and you want to make a note of which commit that was
+for a future date by giving it the tag "tasty".  You can do this like so:
+
+~~~
+$ git tag tasty [commit hash]
+~~~
+
+In my case, I ran:
+
+~~~
+$ git tag tasty 366f4b51e661fd96a9bde001138886a4216b7da1
```

Comment:
Generally, we use the short versions of the tags. I assume that works also here, right?

Reply:
Yup! Done